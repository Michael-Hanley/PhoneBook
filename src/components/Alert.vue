<template>
  <Modal @onClose="closeHandler">
    <div class="modal-style">
      <div>
        <slot></slot>
      </div>
      <div class="button-row">
        <Button :onClick="closeHandler" class="close-button">
          {{ close }}
        </Button>
        <Button :onClick="okHandler"> {{ ok }} </Button>
      </div>
    </div>
  </Modal>
</template>

<script>
import Modal from "./Modal";
import Button from "./Button";
import strings from "@/assets/strings";

export default {
  name: "AddContactModal",
  components: {
    Modal,
    Button,
  },
  props: {
    onClose: Function,
    onOk: Function,
  },
  data: function () {
    return {
      close: strings.close,
      ok: strings.ok,
    };
  },
  methods: {
    closeHandler: function () {
      if (this.onClose) this.onClose();
      this.$el.parentNode.removeChild(this.$el);
    },
    okHandler: function () {
      if (this.onOk) this.onOk();
      this.closeHandler();
    },
  },
};
</script>

<style scoped>
.modal-style {
  min-width: 300px;
  padding: 20px;
}
.close-button {
  background-color: white;
  border: gray;
  color: black;
  margin-right: 10px;
}
.button-row {
  margin-top: 20px;
  text-align: right;
}
</style>



