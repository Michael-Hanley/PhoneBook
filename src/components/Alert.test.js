import { shallowMount } from '@vue/test-utils';
import Alert from './Alert.vue';

let wrapper;
describe('Alert', () => {
    beforeEach(() => {
        wrapper = shallowMount(Alert, {
            slots: {
                default: 'test'
            },
            propsData: {
                onOk: jest.fn(),
                onClose: jest.fn()
            }
        })
    });
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});