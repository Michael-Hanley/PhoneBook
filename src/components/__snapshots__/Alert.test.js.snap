// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render correctly 1`] = `
<modal-stub>
  <div class="modal-style">
    <div>test</div>
    <div class="button-row">
      <button-stub onclick="function () { [native code] }" class="close-button">
        Close
      </button-stub>
      <button-stub onclick="function () { [native code] }"> Ok </button-stub>
    </div>
  </div>
</modal-stub>
`;
